akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -459,7 +459,12 @@ private[persistence] trait Eventsourced extends Snapshotter with PersistenceStas
     }
 
     private val recoveryBehavior: Receive = {
-      val _receiveRecover = receiveRecover
+      val _receiveRecover = try receiveRecover catch {
+        case NonFatal(e) ⇒
+          try onRecoveryFailure(e, Some(e))
+          finally context.stop(self)
```

Comment:
Call `returnRecoveryPermit()` here.

Reply:
When looking at the code in Eventsourced.scala I think there are a few more places where we stop without returning the permit. I will follow up on that in a separate PR.